app/grants/management/commands/subminer.py

```diff
@@ -126,10 +172,57 @@ def handle(self, *args, **options):
                 next_contribution_date__lt=timezone.now(),
                 num_tx_processed__lt=F('num_tx_approved')
             )
-            logger.info(" - %d has %d subs ready for execution", grant.pk, subs.count())
+            if subs.count() > 0:
+                logger.info(" - %d has %d subs ready for execution", grant.pk, subs.count())
 
             for subscription in subs:
                 try:
                     process_subscription(subscription, live)
                 except Exception as e:
                     logger.exception(e)
+
+        unconfirmed_grant_deploy = Grant.objects.filter(
+            contract_address='0x0',
+            deploy_tx_confirmed=False,
+            network=network
+        )
+        logger.info("got %d unconfirmed deploy_grant_txs", unconfirmed_grant_deploy.count())
+
+        for grant in unconfirmed_grant_deploy:
+            listen_for_tx(grant, None, grant.deploy_tx_id, grant.network, 'grant_deploy')
+
+        unconfirmed_grant_cancel = Grant.objects.filter(
+            cancel_tx_confirmed=False,
+            network=network
+        ).exclude(cancel_tx_id='0x0').exclude(cancel_tx_id='')
+        logger.info("got %d unconfirmed cancel_grant_txs", unconfirmed_grant_cancel.count())
+
+        for grant in unconfirmed_grant_cancel:
+            listen_for_tx(grant, None, grant.cancel_tx_id, network, 'grant_cancel')
+
+        unconfirmed_new_approve = Subscription.objects.filter(
+            new_approve_tx_confirmed=False,
+            network=network
+        ).exclude(new_approve_tx_id='0x0').exclude(new_approve_tx_id='')
+        logger.info("got %d unconfirmed new_approve_txs", unconfirmed_new_approve.count())
+
+        for sub in unconfirmed_new_approve:
+            listen_for_tx(None, sub, sub.new_approve_tx_id, network, 'new_approve')
+
+        unconfirmed_end_approve = Subscription.objects.filter(
+            end_approve_tx_confirmed=False,
```

Comment:
what if there is no end approve tx?  we need to further drill down on this queryset

Reply:
If there is no end_approve_tx then there is nothing for the subminer to process. I dont think there is any needed extra work here.